tests/base/consensus_algorithm_01.cc

```diff
@@ -0,0 +1,71 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2020 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+// Test AnonymousConsensusAlgorithmProcess.
+
+#include <deal.II/base/mpi_consensus_algorithm.h>
+
+#include "../tests.h"
+
+using namespace dealii;
+
+void
+test(const MPI_Comm &comm)
+{
+  const unsigned int my_rank = dealii::Utilities::MPI::this_mpi_process(comm);
+  const unsigned int n_rank  = dealii::Utilities::MPI::n_mpi_processes(comm);
+
+  dealii::Utilities::MPI::AnonymousConsensusAlgorithmProcess<
+    dealii::types::global_dof_index,
+    unsigned int>
+    process(
+      [&]() {
+        std::vector<unsigned int> result{(my_rank + 1) % n_rank};
+        return result;
+      },
+      [&](const auto other_rank, auto &send_buffer) {
```

Comment:
```suggestion
      [&](const unsigned int other_rank, std::vector<dealii::types::global_dof_index> &send_buffer) {
```
and similarly below.